legend-engine-xt-persistence-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/test/assertion/allRowsEquivalentToJson/AllRowsEquivalentToJsonAssertionParserGrammar.g4

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+parser grammar AllRowsEquivalentToJsonAssertionParserGrammar;
+
+import CoreParserGrammar;
+
+options
+{
+    tokenVocab = AllRowsEquivalentToJsonAssertionLexerGrammar;
+}
+
+
+// -------------------------------------- IDENTIFIER -------------------------------------
+
+identifier:                            VALID_STRING | STRING | ALL_ROWS_EQUIVALENT_TO_JSON | EXPECTED
+;
+
+
+// -------------------------------------- DEFINITION -------------------------------------
+
+definition:                            expectedDefinition
+                                       EOF
+;
+expectedDefinition:                    EXPECTED COLON embeddedData SEMI_COLON
+;
+embeddedData:                          identifier ISLAND_OPEN (embeddedDataContent)*
+;
+embeddedDataContent:                   ISLAND_START | ISLAND_BRACE_OPEN | ISLAND_CONTENT | ISLAND_HASH | ISLAND_BRACE_CLOSE | ISLAND_END
```

Comment:
hmm do you think having these parser statements in common file which can be included would be better? I see these are just replica's of EqualToJson?

Reply:
(Similar comment as to the one for the new assert fail classes)
We have had some internal discussion, and here is what we think: as the two new assertion types are essentially different logical entities from the existing EqualToJson, we feel we should not try to reuse or try to establish a relationship with EqualToJson. We shall take it that at the moment it happens to be the case that ActiveRowsEquivalentToJson looks like EqualToJson, but it is entirely possible that it evolves into something different, hence a new class.

Reply:
I think trying to think this could be different than it is currently is a mental trap that could lead to code that will not add value until/unless we actually make it different.

Still, even further, making future changes that could make this substantially different means address breaking changes, and that should be a NO-NO 99.99% of the time.  

If we have use-cases in mind that the current implementation don't meet their needs, what are those?  But if we don't have any use-cases, lets avoid the mental trap that we might need to change these in the future.